
#### Next-Gen Seq with Breast Caner

[__Next generation sequencing in breast cancer: first take home messages__](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3713550/)    First, the sequencing of a few hundreds of breast tumors has revealed new cancer genes. Although these were not frequently mutated, mutated genes from different patients could be grouped into the deregulation of similar pathways. Second, NGS allowed further exploration of intratumor heterogeneity and revealed that although subclonal mutations were present in all tumors, there was always a dominant clone which comprised at least 50% of the tumor cells. Finally, tumor-specific DNA rearrangements could be detected in the patient’s plasma, suggesting that NGS could be used to personalize the monitoring of the disease.

[__Transcriptomic landscape of breast cancers through mRNA sequencing__](http://www.nature.com/articles/srep00264)   
A really good paper with detailed methods, could be really useful! 17 case with TNBC (basal), non-TNBC, and HER2-positive (both luminal) yields 1.2 billion raw reads. Ientified common transcriptional regulatory elements, such as highly abundant primary transcripts, including osteonectin, RACK1, calnexin, calreticulin, FTL, and B2M, and “genomic hotspots” enriched in primary transcripts between the three groups.

![rnaseq pipline](http://www.nature.com/article-assets/npg/srep/2012/120214/srep00264/images_hires/w926/srep00264-f1.jpg)  

[__RNA sequencing of cancer reveals novel splicing alterations__](http://www.nature.com/articles/srep01689)  
Splicing signature on TNBC, non-TNBC, and HER2-positive breast cancer compare to human breast organoids (epithelium) samples derived from normal healthy women, a continuation of the previous paper. Discovered differentially spliced genes and splice isoforms. Also found that exon skip and intron retention are predominant splice events in breast cancer. Validated the presence of novel hybrid isoforms of critical molecules like CDK4, LARP1, ADD3, and PHLPP2.

![rnaseq splice variant pipeline](http://www.nature.com/article-assets/npg/srep/2013/130422/srep01689/images_hires/w926/srep01689-f1.jpg)

[_Whole Transcriptome RNA-Seq Analysis of Breast Cancer Recurrence Risk Using Formalin-Fixed Paraffin-Embedded Tumor Tissue_](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0040092#s2)  
Focused on RNAseq for FFPE tissues. Writing style may be useful.

[_Identification of fusion genes in breast cancer by
paired-end RNA-sequencing_](http://download.springer.com/static/pdf/529/art%253A10.1186%252Fgb-2011-12-1-r6.pdf?originUrl=http%3A%2F%2Fgenomebiology.biomedcentral.com%2Farticle%2F10.1186%2Fgb-2011-12-1-r6&token2=exp=1463503611~acl=%2Fstatic%2Fpdf%2F529%2Fart%25253A10.1186%25252Fgb-2011-12-1-r6.pdf%2A~hmac=254e1f6c59c7df92a4fa6341a3b4a1d4d32cf0ad03a6c9e5e9201ca8f6ec6720)  
Maybe useful if wants to determine gene fusion with RNAseq data.



---

#### Single Cell sequencing
[__Single-cell exome sequencing reveals single-nucleotide mutation characteristics of a kidney tumor__](http://www.ncbi.nlm.nih.gov/pubmed/22385958)  
Tumor evolution.  The heterogeneous nature of tumors, however, makes it difficult for researchers to analyze the intratumoral genetic structure and identify key changes during neoplasm progression. Mostly developing single-cell sequencing technologies.


---

#### Third Generation Sequencing
[__A window into third-generation sequencing__](http://www.ncbi.nlm.nih.gov/pubmed/22385958)  
This loss of synchronicity (called _dephasing_) causes an increase in noise and sequencing errors as the read extends, effectively limiting the read length produced by the most widely used SGS systems to significantly less than the average read lengths achieved by Sanger sequencing. Third-generation DNA-sequencing technologies are distinguished by direct inspection of single molecules with methods that do not require wash steps during DNA synthesis.
